clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -198,6 +208,41 @@ public boolean hasCompletedFetches() {
         return !completedFetches.isEmpty();
     }
 
+    /**
+     * Return if there are buffered CompletedFetches for a paused TopicPartition.
+     *
+     * @param tp {@link TopicPartition}
+     * @return  {@code true} if there are buffered {@link CompletedFetch},  {@false} otherwise
+     */
+    public boolean hasPausedCompletedFetchesFor(TopicPartition tp) {
```

Comment:
Can you make this method package private by removing the keyword `public` and add comment that this method is used by test only?

Reply:
changed it to. : hasBufferedDataForPausedPartition(). I was thinking if there will be an usecase for apps to query if there is any buffered data for a paused partitions. I currently can't think of any. So we can make it package private for now and think of exposing it if there is a need in future.